lib/fog/azurerm/models/network/network_security_group.rb

```diff
@@ -48,6 +48,32 @@ def destroy
           service.delete_network_security_group(resource_group, name)
         end
 
+        def update(hash = {})
+          raise('Provided hash can not be empty.') if hash.empty? || hash.nil?
+
+          if !hash[:name].nil? || !hash[:location].nil? || !hash[:resource_group].nil? || !hash[:id].nil?
+            raise('You can not modify id:, name:, location: or resource_group:')
+          end
+          unless hash[:security_rules].nil?
+            validate_security_rules(hash[:security_rules])
+            merge_attributes(hash)
+            nsg = service.create_network_security_group(resource_group, name, location, security_rules)
+            return merge_attributes(Fog::Network::AzureRM::NetworkSecurityGroup.parse(nsg))
+          end
+          raise 'Invalid hash key.'
+        end
+
+        def add_security_rules(resource_group, security_group_name, security_rules)
+          validate_security_rules(security_rules)
+          nsg = service.add_security_rules(resource_group, security_group_name, security_rules)
```

Comment:
Please confirm if add_security_rules method updates the original object or creates a new one